File: clientside/src/app/entities/workplan.ts

import {Device} from "src/app/entities/device";
import { Instrukcija } from "./instrukcija";

export class Workplan {
    id:number;
    type:String;
    order:String;
    incidentId:number;
    status:String;
    startDate:Date;
    endDate:Date;
    crewId:number;
    createdBy:String;
    purpuse:String;
    notes:String;
    company:String;
    phoneNumber:String;
    createdOn:Date;
    devices: String;
    instructions:String;
    multimedia:String;

    constructor(type:String,status:String,startDate:Date,endDate:Date,crewId:number,purpuse:String,company:String,order?:String,incidentId?:number,notes?:String,phoneNumber?:String,multimedia?:String,instructions?:String,devices?:String){
        this.type=type;
        this.order=order;
        this.incidentId=incidentId;
        this.status=status;
        this.startDate=startDate;
        this.endDate=endDate;
        this.crewId=crewId;
        this.createdBy=localStorage.getItem("sessionUsername");
        this.purpuse=purpuse;
        this.notes=notes;
        this.company=company;
        this.phoneNumber=phoneNumber;
        this.createdOn=new Date(Date.now());
        this.createdOn.setHours(this.createdOn.getHours()+2);
        this.devices=devices;
        this.instructions=instructions;
        this.multimedia=multimedia;
    }
}

File: Core/SmartGrid/Models/Workplan.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;

namespace SmartGrid.Models
{
    public class Workplan
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int Id { get; set; }

        [Required]
        public string Type { get; set; }

        public string Order { get; set; }

        public int IncidentId { get; set; }

        [Required]
        public string Status { get; set; }

        [Required]
        public string StartDate { get; set; }

        [Required]
        public string EndDate { get; set; }

        [Required]
        public int CrewId { get; set; }

        [Required]
        public string CreatedBy { get; set; }

        [Required]
        public string Purpuse { get; set; }

        public string Notes { get; set; }

        [Required]
        public string Company { get; set; }

        public string PhoneNumber { get; set; }
        [Required]
        public DateTime CreatedOn { get; set; }

        public string Devices { get; set; }

        public string Instructions { get; set; }

        public string Multimedia { get; set; }
    }
}
